File: typescript/tests/wire/executionObjectives.test.ts

import * as Sikaru from "../../src/api/index";
import { SikaruApi } from "../../src/Client";
import { mockServerPool } from "../mock-server/MockServerPool";

describe("ExecutionObjectivesClient", () => {
    
    test("list_objectives (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-objectives").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionObjectives.listObjectives("project_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("list_objectives (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-objectives").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionObjectives.listObjectives("project_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("create (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "criterion" : "criterion" , "idempotency_key" : "idempotency_key" , "objective" : "objective" , "run_id" : "run_id" , "session_id" : "session_id" };
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-objectives").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionObjectives.create("project_id", {
    criterion: "criterion",
    idempotency_key: "idempotency_key",
    objective: "objective",
    run_id: "run_id",
    session_id: "session_id"
});
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("create (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { "criterion" : "x" , "idempotency_key" : "x" , "objective" : "x" , "run_id" : "x" , "session_id" : "x" };
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-objectives").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionObjectives.create("project_id", {
    criterion: "x",
    idempotency_key: "x",
    objective: "x",
    run_id: "x",
    session_id: "x"
})
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("get (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-objectives/objective_id").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionObjectives.get("project_id", "objective_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("get (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .get("/v1/projects/project_id/execution-objectives/objective_id").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionObjectives.get("project_id", "objective_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("cancel (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-objectives/objective_id/cancel").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionObjectives.cancel("project_id", "objective_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("cancel (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-objectives/objective_id/cancel").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionObjectives.cancel("project_id", "objective_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("pause (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-objectives/objective_id/pause").respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionObjectives.pause("project_id", "objective_id");
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("pause (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-objectives/objective_id/pause").respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionObjectives.pause("project_id", "objective_id")
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
    test("resume (1)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = { };
        const rawResponseBody = { "key" : "value" };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-objectives/objective_id/resume").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(200).jsonBody(rawResponseBody)
                .build();

        
                        
                                const response = await client.executionObjectives.resume("project_id", "objective_id", {});
                                expect(response).toEqual(rawResponseBody);
                              
                    
    });
          
    test("resume (2)", async () => {
        const server = mockServerPool.createServer();
        const client = new SikaruApi({ "maxRetries" : 0 , "apiKey" : "test" , "environment" : server.baseUrl });
        const rawRequestBody = null;
        const rawResponseBody = { };
        
        server
            .mockEndpoint()
            .post("/v1/projects/project_id/execution-objectives/objective_id/resume").jsonBody(rawRequestBody)
                .respondWith()
            .statusCode(422).jsonBody(rawResponseBody)
                .build();

        
            await expect(async () => {
                return await client.executionObjectives.resume("project_id", "objective_id", null)
            }).rejects.toThrow(Sikaru.UnprocessableEntityError);
    });
          
});

File: ruby/test/wire/execution_objectives_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class ExecutionObjectivesWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_execution_objectives_list_objectives_with_wiremock
    test_id = "execution_objectives.list_objectives.0"

    @client.execution_objectives.list_objectives(
      project_id: "project_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_objectives.list_objectives.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/execution-objectives",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_objectives_create_with_wiremock
    test_id = "execution_objectives.create.0"

    @client.execution_objectives.create(
      project_id: "project_id",
      criterion: "criterion",
      idempotency_key: "idempotency_key",
      objective: "objective",
      run_id: "run_id",
      session_id: "session_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_objectives.create.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/execution-objectives",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_objectives_get_with_wiremock
    test_id = "execution_objectives.get.0"

    @client.execution_objectives.get(
      project_id: "project_id",
      objective_id: "objective_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_objectives.get.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/execution-objectives/objective_id",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_objectives_cancel_with_wiremock
    test_id = "execution_objectives.cancel.0"

    @client.execution_objectives.cancel(
      project_id: "project_id",
      objective_id: "objective_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_objectives.cancel.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/execution-objectives/objective_id/cancel",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_objectives_pause_with_wiremock
    test_id = "execution_objectives.pause.0"

    @client.execution_objectives.pause(
      project_id: "project_id",
      objective_id: "objective_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_objectives.pause.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/execution-objectives/objective_id/pause",
      query_params: nil,
      expected: 1
    )
  end

  def test_execution_objectives_resume_with_wiremock
    test_id = "execution_objectives.resume.0"

    @client.execution_objectives.resume(
      project_id: "project_id",
      objective_id: "objective_id",
      request: {},
      request_options: {
        additional_headers: {
          "X-Test-Id" => "execution_objectives.resume.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/execution-objectives/objective_id/resume",
      query_params: nil,
      expected: 1
    )
  end
end
